### PR TITLE
[29] Create React frontend replicating JQuery frontend

### LINKED ISSUE #29: Convert existing frontend to React

### PR DESCRIPTION
**Pull request by TaylorAbraham**
Fixes #29 
image

### CODE REVIEW

File: src/main/js/App.js [hunk 1]
```diff
@@ -1,11 +1,143 @@
...
+                <select id="questionType">
+                    <option value={qType.OPEN_ENDED} onChange={setCurrentType(qType.OPEN_ENDED)}>OpenEnded</option>
+                </select>
+                <Button variant="contained" color="primary" onClick={addQuestion}>+</Button>
+                {questions.map((q, i) => {
+                    switch(q.type) {
+                        case qType.OPEN_ENDED:
+                            return (
+                                <div key={i}>
+                                    <input
```
Comment (MichaelDysart) [inline src/main/js/App.js]: We seem to have lost the ability to add questions in the middle of the question list and the ability to remove questions.

**Review by MichaelDysart — CHANGES_REQUESTED** [review]
Looks good. Just see if it is possible to add the + and - buttons to inputs for the question divs.

File: src/main/js/App.js (hunk 1 above)
Comment (TaylorAbraham) [reply src/main/js/App.js]: Good catch, on it

File: src/main/js/App.js [hunk 2]
```diff
@@ -1,11 +1,259 @@
...
+                    <InputLabel id="qtype_select_label">Question Type</InputLabel>
+                    <Select labelId="qtype_select_label" value={currentType} onChange={e => setCurrentType(e.target.value)}>
+                        <MenuItem value={qType.OPEN_ENDED}>Open-Ended</MenuItem>
+                        <MenuItem value={qType.NUMERICAL}>Numerical</MenuItem>
+                    </Select>
+                </FormControl>
+                <Button className="qq-app m" variant="contained" color="primary" disabled={!currentType} onClick={addQuestion}>+</Button>
+            </div>
+            <div>
+                <Button className="qq-app mv" variant="contained" color="primary" onClick={myFunc1}>Dev Test 1 (POST)</Button>
```
Comment (Fahid-Mannan) [inline src/main/js/App.js]: oh no "myFunc1" has persisted from the other PR...

File: src/main/js/App.js [hunk 3]
```diff
@@ -1,11 +1,259 @@
...
+                                    <Button
+                                        className="qq-app m"
+                                        variant="contained"
+                                        color="primary"
+                                        onClick={() => deleteQuestion(i)}
+                                    ><DeleteIcon /></Button>
+                                </div>
+                            );
+                        case qType.NUMERICAL:
+                            // @TODO: Add min/max
```
Comment (Fahid-Mannan) [inline src/main/js/App.js]: Hasn't min/max already been added?

File: src/main/js/App.js (hunk 2 above)
Comment (TaylorAbraham) [reply src/main/js/App.js]: The discussion ended with us leaving the naming intact for the time being, as this is a developer test function.

File: src/main/js/App.js (hunk 3 above)
Comment (TaylorAbraham) [reply src/main/js/App.js]: Just caught that too haha. I removed that comment in the next commit.

**Comment by Fahid-Mannan** [comment]
Looks good, we can start implementing our project topic into code now!

**Review by rinadina — APPROVED** [review]
Looks fine to me.

**Review by MichaelDysart — APPROVED** [review]

**Pull request merged**